### DISCUSSION TITLE
AKS addon azure-keyvault-secrets-provider

### DISCUSSION THREAD

**Discussion by schwichti** [Azure/secrets-store-csi-driver-provider-azure#795] [Q&A] [answered]
What is the relation/difference between using the CSI driver from this repository vs. the AKS addon `azure-keyvault-secrets-provider?`

You can install the addon with this command:
```bash
az aks create -n myAKSCluster -g myResourceGroup --enable-addons azure-keyvault-secrets-provider --enable-managed-identity
```

See https://docs.microsoft.com/en-us/azure/aks/csi-secrets-store-driver

**Comment by aramase** [comment, marked as answer]
Hello 👋🏻  There is no difference between the 2. The AKS add-on is a managed offering of this project. AKS handles the deployment and upgrade of the driver and provider when enabled as an add-on. 

If you install the driver and provider manually using helm charts from OSS, it's not officially supported by Microsoft: https://github.com/Azure/secrets-store-csi-driver-provider-azure#support. For OSS, support is only via GitHub issues. Instead If you enable the add-on, it is supported and you can open support tickets if you have any issues.
